backend/src/main/java/com/ad1/loggenerator/service/implementation/AWSStreamService.java

```diff
@@ -183,6 +183,8 @@ public void streamToS3(SelectionModel selectionModel, StreamTracker streamJobTra
         // Count the log lines saved to the bucket object, set url and the log count to the stream job tracker
         streamJobTracker.setLogCount(awsLogService.getLogCount(s3Client, s3Object, bucketName, key));
         streamJobTracker.setStreamObjectURL(objectURL);
+        if (streamJobTracker.getStreamObjectURL() == null)
+            streamJobTracker.setStatus(JobStatus.FAILED);
```

Comment:
This check should be moved higher up, so we can skip initializing the s3Client if it fails

Reply:
Okay, thanks. I will do that.